src/adapter/src/coord/sequencer/inner/peek.rs

```diff
@@ -609,6 +620,30 @@ impl Coordinator {
                 let stage = match pipeline() {
                     Ok(Either::Left(global_lir_plan)) => {
                         let optimizer = optimizer.unwrap_left();
+                        // Enable fast path cluster calculation for slow path plans.
+                        let mut needs_plan_insights = explain_ctx.needs_plan_insights();
+                        let opt_limit = mz_adapter_types::dyncfgs::PLAN_INSIGHTS_NOTICE_FAST_PATH_CLUSTERS_OPTIMIZE_DURATION.get(catalog.system_config().dyncfgs());
+                        // However, disable this if we only want the notice and plan optimization
+                        // took longer than the threshold. This is to prevent a situation where
+                        // optimizing takes a while and there a lots of clusters, which would delay
+                        // peek execution by the product of those.
+                        if matches!(explain_ctx, ExplainContext::PlanInsightsNotice(_))
+                            && optimizer.duration() > opt_limit
+                        {
+                            needs_plan_insights = false;
```

Comment:
This disables the entire plan insights machinery? I thought only the `compute_fast_path_clusters` insight would need to be disabled.

Reply:
Good catch! I'll plumb this down so it only restricts the slow parts.